### PR TITLE
Add for chars and XML

### PR DESCRIPTION
**Pull request by thehabes**
For #553 and #554 

This is when a user clicks Add a Button or Add a Tag on the interface.  Now, the default button is in the database in the max position + 1 after being created.  Next will be need to fix the "Save Changes" functionality.

### CODE REVIEW

File: src/java/servlets/addTag.java [hunk 1]
```diff
@@ -39,28 +39,26 @@
...
-                        while (new TagButton(projectID, ctr, true).exists()) {
-                            ctr++;
-                        }
-                        TagButton thisTag=new TagButton(projectID, ctr, "new", false, "description");
-                        out.print(""+ctr);
-                        return;
-
-                    }
+                int projectID=parseInt(request.getParameter("projectID"));
+                position = TagButton.getMaxPosition(projectID) + 1;
```
Comment (cubap) [inline src/java/servlets/addTag.java:57]: It's not perfect, but this is a much shinier turd.

**Review by cubap — APPROVED** [review]
great work.

File: src/java/edu/slu/tpen/servlet/SaveNewProjectXML.java [hunk 2]
```diff
@@ -54,7 +55,7 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         PrintWriter out = response.getWriter();
         String buttonToSendOut = "";
         try {
-            TagButton newButton = new TagButton(projectID, position, tag, true, description);
+            TagButton newButton = new TagButton(projectID, position, tag, description, true);
```
[the remark this answers is not in the document]
Comment (thehabes) [reply src/java/edu/slu/tpen/servlet/SaveNewProjectXML.java:58]: Friggin builds, caring about my waste code.

File: src/java/servlets/addHotkey.java [hunk 3]
```diff
@@ -39,28 +39,26 @@
...
-                        int ctr = 1;
-                        int UID=parseInt(request.getParameter("UID"));
-                        while (new Hotkey(UID, ctr).exists()) {
-                            ctr++;
-                        }
-                        new Hotkey(110,UID, ctr);
-                        out.print(""+ctr);
-                        return;
-                    }
+                //FIXME if this is actually used, it also does the position wrong.
```
[the remark this answers is not in the document]
Comment (thehabes) [reply src/java/servlets/addHotkey.java:45]: Yep, that's why I didn't blow it away.  I figured it was used somewhere, prolly like transcription.jsp which is alive.

File: src/java/servlets/addTag.java (hunk 1 above)
Comment (thehabes) [reply src/java/servlets/addTag.java:57]: Yep.  Not replacing this module, so gotta work with the crap.

**Pull request merged**